python/rikai/torch/data.py

```diff
@@ -254,3 +268,55 @@ def __iter__(self) -> Generator[Dict, None, None]:
             stop.set()  # tell the prefetch thread to stop
         prefetch_thd.join()
         q.join()
+
+
+def _maybe_cache_df(dataset_ref: Union[str, Path, 'pyspark.sql.DataFrame'],
+                    parquet_row_group_size_bytes: int) -> str:
+    """
+    If the given dataset_ref is a str/Path, then just return a str ref.
+    If it's a pyspark DataFrame then cache the DataFrame as parquet and
+    return the cache uri
+
+    Parameters
+    ----------
+    dataset_ref: str, Path, or pyspark DataFrame
+        Either a uri to parquet or a DataFrame that will be cached as parquet
+    parquet_row_group_size_bytes: int
+        Row group size in bytes when writing to parquet
+
+    Returns
+    -------
+    uri: str
+        Either the input parquet uri or the cached uri for a DataFrame
+    """
+    if isinstance(dataset_ref, (str, Path)):
+        return str(dataset_ref)
+    else:
+        from pyspark.sql import DataFrame
+        from rikai.spark.utils import df_to_parquet
+        if isinstance(dataset_ref, DataFrame):
+            kwargs = {}
+            if parquet_row_group_size_bytes is not None:
+                kwargs['parquet_row_group_size_bytes'] = \
+                    parquet_row_group_size_bytes
+            _get_cache_uri(dataset_ref)
+            df_to_parquet(dataset_ref, cache_uri, **kwargs)
```

Comment:
raise if it is not a dataframe, or pyspark is not imported / installed?

Reply:
added more checks